### DISCUSSION TITLE
std::time vs tokio::time

### DISCUSSION THREAD

**Discussion by Banyc** [tokio-rs/tokio#4633] [Q&A] [answered]
What is the difference between them and which one should I use? I cannot find answer elsewhere.

Can we reduce duplicates?

**Comment by Darksonn** [comment, marked as answer]
The sleeping functions in the standard library are blocking, so they can't be used in async code. The ones in `tokio::time` are the alternatives. You can read more about the difference between blocking and async functions [here][1].

[1]: https://ryhl.io/blog/async-what-is-blocking/

**Comment by YarekTyshchenko** [comment]
Came here to find out why `Instant` was duplicated? Why should I use Tokio's Instant when `std` one looks like does the same thing?

**Reply by Darksonn** [reply]
That's because Tokio supports pausing time. The Tokio `Instant` type will remain constant when time is paused, but the std `Instant` will still advance. See [`tokio::time::pause`](https://docs.rs/tokio/latest/tokio/time/fn.pause.html) for more information.
